File: 윤효전/211019/2456.js
const stdin = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : 'input.txt').toString().split('\n');
const input = (() => {
    let line = 0
    return () => stdin[line++]
})();

N = Number(input())

const candidates = [...Array(3)].map((v, i) => new Map([['1', 0], ['2', 0], ['3', 0], ['total', 0], ['num', i + 1]]))

for (let i = 0; i < N; i++) {
    const [a, b, c] = input().trimRight().split(' ')
    candidates[0].set(a, candidates[0].get(a) + 1)
    candidates[0].set('total', candidates[0].get('total') + Number(a))
    candidates[1].set(b, candidates[1].get(b) + 1)
    candidates[1].set('total', candidates[1].get('total') + Number(b))
    candidates[2].set(c, candidates[2].get(c) + 1)
    candidates[2].set('total', candidates[2].get('total') + Number(c))
}

candidates.sort((a, b) => {
    if (a.get('total') == b.get('total')) {
        if (a.get('3') == b.get('3')) {
            return a.get('2') - b.get('2')
        }
        return a.get('3') - b.get('3')
    }
    return a.get('total') - b.get('total')
})

const [a, b] = [candidates[2], candidates[1]]
if (a.get('total') != b.get('total')) {
    console.log(a.get('num'), a.get('total'))
}
else if (a.get('1') != b.get('1')) {
    console.log(a.get('num'), a.get('total'))
}
else {
    console.log(0, a.get('total'))
}

File: 윤효전/211019/2456.py
import sys
sys.stdin = open('input.txt')
input = sys.stdin.readline

N = int(input())
candidates = [{'1': 0, '2': 0, '3': 0, 'total': 0, 'num': i}
              for i in range(1, 4)]

for _ in range(N):
    a, b, c = input().rstrip().split()
    candidates[0][a] += 1
    candidates[0]['total'] += int(a)
    candidates[1][b] += 1
    candidates[1]['total'] += int(b)
    candidates[2][c] += 1
    candidates[2]['total'] += int(c)

candidates.sort(key=lambda x: (x['total'], x['3'], x['2']))
a, b = candidates[-1], candidates[-2]
if a['total'] != b['total']:
    print(a['num'], a['total'])
elif a['1'] != b['1']:
    print(a['num'], a['total'])
else:
    print(0, a['total'])
